rating/peersRatingHandler.go

```diff
@@ -129,10 +183,88 @@ func (prh *peersRatingHandler) getOldRating(pid core.PeerID) (int32, bool) {
 	return defaultRating, found
 }
 
-func (prh *peersRatingHandler) updateRatingIfNeeded(pid core.PeerID, updateFactor int32) {
-	defer prh.updateMetrics()
+func (prh *peersRatingHandler) markInactivePIDsForRemoval(receivedPIDs map[string]struct{}) {
+	removalTimestamp := prh.getTimeHandler().Add(prh.timeWaitingForReconnection).Unix()
+	topRatedPIDs := prh.topRatedCache.Keys()
+	for _, pid := range topRatedPIDs {
+		_, isPIDStillActive := receivedPIDs[string(pid)]
+		alreadyMarked := prh.markedForRemovalCache.Has(pid)
+		if !isPIDStillActive && !alreadyMarked {
+			prh.markedForRemovalCache.Put(pid, removalTimestamp, int64Size)
+		}
+	}
+	badRatedPIDs := prh.badRatedCache.Keys()
+	for _, pid := range badRatedPIDs {
+		_, isPIDStillActive := receivedPIDs[string(pid)]
+		alreadyMarked := prh.markedForRemovalCache.Has(pid)
+		if !isPIDStillActive && !alreadyMarked {
+			prh.markedForRemovalCache.Put(pid, removalTimestamp, int64Size)
+		}
+	}
+}
+
+func (prh *peersRatingHandler) sweepCachersLoop(ctx context.Context) {
```

Comment:
can merge this function with the updateMetricsLoop as follows:
```GO
func (prh *peersRatingHandler) processLoop(ctx context.Context){
	timerCachersSweep := time.NewTimer(prh.timeBetweenCachersSweep)
	timerMetricsUpdate := time.NewTimer(prh.timeBetweenMetricsUpdate)
	
	defer timerCachersSweep.Stop()
	defer timerMetricsUpdate.Stop()

	for {
		select {
		case <-ctx.Done():
			return
		case <-timerCachersSweep.C:
			prh.sweepCachers()
			timerCachersSweep.Reset(prh.timeBetweenCachersSweep)
		case <-timerMetricsUpdate.C:
			prh.updateMetrics()
			timerMetricsUpdate.Reset(prh.timeBetweenMetricsUpdate)
		}
	}
}

```

Reply:
updated